lib/src/solver/result.dart

```diff
@@ -109,6 +122,34 @@ class SolveResult {
     }
   }
 
+  /// Send analytics about the package resolution.
+  void sendAnalytics(Analytics analytics) {
+    ArgumentError.checkNotNull(analytics);
+
+    for (final package in packages) {
+      final source = package.source;
+      // Only send analytics for packages from pub.dev.
+      if (source is HostedSource &&
+          (runningFromTest ||
+              package.description['url'] == HostedSource.pubDevUrl)) {
+        final dependencyType = const {
+          DependencyType.dev: 'dev',
+          DependencyType.direct: 'direct',
+          DependencyType.none: 'transitive'
+        }[_root.dependencyType(package.name)];
+
+        analytics.sendEvent(
+          'pub-get',
```

Comment:
Should this be `pub.dev hit` or `pub-hit` or `pub-package-hit` or maybe: `pub-dev-hit`, that avoids dots/space and signals that this is a `pub.dev` thing -- since we don't count hits for other package repositories.